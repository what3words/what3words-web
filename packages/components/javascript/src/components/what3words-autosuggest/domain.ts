export interface SDKQueryParams {
  key?: string;
  baseUrl?: string;
  headers?: Record<string, string>;
  callback?: string;
}

export interface AutosuggestOptionsProps {
  open: boolean;
  options?: AutosuggestOption[];
  option?: AutosuggestOption;
}

export interface AutosuggestOption {
  country: string;
  language: string;
  nearestPlace: string;
  rank: number;
  words: string;
  distanceToFocusKm?: number;
}

export interface CustomOption {
  value: string;
  id: string;
  description?: string;
  distance?: {
    value: number;
    units?: string;
  };
}

export interface InputValuePayload {
  value: string;
}

export interface CoordinatesPayload {
  coordinates: { lat: number; lng: number };
}

export interface SuggestionPayload {
  suggestion: AutosuggestOption;
}

export interface SuggestionsPayload {
  suggestions: AutosuggestOption[];
}
