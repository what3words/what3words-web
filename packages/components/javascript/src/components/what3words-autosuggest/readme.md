# what3words-autosuggest

<!-- Auto Generated Below -->


## Usage

### Angular

#### What3Words Autosuggest Component

##### Installation

```bash
npm install @what3words/angular-components@<PACKAGE-VERSION>
```

##### Usage

```js
import { Component } from "@angular/core";
import { RouterOutlet } from "@angular/router";

import { What3wordsAutosuggest } from "@what3words/angular-components";

@Component({
  selector: "app-root",
  imports: [RouterOutlet, What3wordsAutosuggest],
  template: `
    <what3words-autosuggest [api_key]="w3w_api_key">
      <input
        id="search-input"
        type="text"
        placeholder="Find your address"
        autocomplete="off"
      />
    </what3words-autosuggest>
    <router-outlet />
  `,
  styles: [``],
})
export class AppComponent {
  w3w_api_key = "<W3W-API-KEY>";
}
```


### Javascript

#### What3Words Autosuggest Component

##### Configuration

###### Script Parameters

When using the CDN to load this package, you can provide query string parameters to the `<script>` tag to pass the following query string parameters to the script loader once it is running:

- `key=<API_KEY>` This will provide the API Key to the SDK in the `<script>`
- `callback=<NAME>` This will provide a callback name for the SDK to call once the script loads. You should attach this function to the global window for this to actually call your script.
- `baseUrl=<URL>` The base url to pass to the SDK to call autosuggest endpoint at. (Useful for mocking in tests or pointing to a different environment.)
- `headers=<STRINGIFIED_JSON>` You can set headers that the SDK will pass to the backend autosuggest API when making requests against the API. (This is useful for tracking purposes or providing specialised agents, for example)

> Navigating to http://localhost:8080?key=API_KEY&callback=NAME&baseUrl=URL&headers={"custom-header":"foo"} will set the above script tag parameters.

####### Component Attributes

Component attributes can also be passed using script parameters. To pass attributes key/value pairs to the autosuggest component you can pass the attribute name prefixed with the component attribute prefix. The prefix defaults to `comp_` so if, for example you wanted to add the `initial-value` attribute to the component, you would provide the query paramter `comp_initial-value=<INITIAL_VALUE>`. You can change the prefix using the query parameter:

- `componentPrefix=<PREFX>` (default is `comp_`) This will change the prefix for any component attributes.
- `comp_<ATTRIBUTE>=<VALUE>` This will set the attribute `<ATTRIBUTE>` on the autosuggest component with value `<VALUE>`. (This assumes that the prefix has not been changed otherwise you should change `comp_` for the `<PREFIX>` you set it to.)

####### Custom HTML Input

If you want to have a custom HTML input being wrapped around by the `<what3words-autosuggest />` custom component tag, you must first enable this behaviour using the query parameter:

- `customInput=true`

You can also provide attributes to the input similarly to the autosuggest component attributes above by providing the attribute with the `inputPrefix` prefixing the attribute key/value pair. For example to set the `id` and `name` attributes on the custom input you would do provide the query parameters `input_id=my-id&input_name=my-form-elem`. You can also change the prefix if you wish by providing the following query paramter:

- `inputPrefix=<PREFIX>` (default is `input_`) This will change the prefix for any input attributes.
- `input_<ATTRIBUTE>=<VALUE>` This will set the attribute `<ATTRIBUTE>` on the input element with value `<VALUE>`. (This assumes that the prefix has not been changed otherwise you should change `input_` for the `<PREFIX>` you set it to.)

> :warning: **WARN**: The custom input only works if you have not disabled the autosuggest component

####### Disable Autosuggest Component

If you want to disable the Autosuggest component tag, so that the page does not add it at all and just adds the SDK which is accessible via `window.what3words` then you can provide the following query paramter to disable to the component:

- `disableAutosuggest=true`

##### Installation

###### CDN

```html
<head>
  <script
    type="module"
    defer
    src="https://cdn.what3words.com/javascript-components@<PACKAGE-VERSION>/dist/what3words/what3words.esm.js"
  ></script>
  <script
    nomodule
    defer
    src="https://cdn.what3words.com/javascript-components@<PACKAGE-VERSION>/dist/what3words/what3words.js"
  ></script>
  ...
</head>
...
```

###### NPM

```bash
npm install @what3words/javascript-components@<PACKAGE-VERSION>
```

##### Usage

```html
<body>
  <what3words-autosuggest api_key="<W3W-API-KEY>">
    <input
      id="search-input"
      type="text"
      placeholder="Find your address"
      autocomplete="off"
    />
  </what3words-autosuggest>
</body>
```


### React

#### What3Words Autosuggest Component

##### Installation

```bash
npm install @what3words/react-components@<PACKAGE-VERSION>
```

##### Usage

```jsx
import { What3wordsAutosuggest } from "@what3words/react-components";

const W3W_API_KEY = "<W3W-API-KEY>";

export default function Autosuggest() {
  return (
    <What3wordsAutosuggest api_key={API_KEY}>
      <input
        type="text"
        placeholder="Find your address"
        style={{ width: "300px" }}
        autoComplete="off"
      />
    </What3wordsAutosuggest>
  );
}
```


### Vue

#### What3Words Autosuggest Component

##### Installation

```bash
npm install @what3words/vue-components@<PACKAGE-VERSION>
```

##### Usage

```jsx
<template>
  <What3wordsAutosuggest :="$props">
    <input
      id="search-input"
      type="text"
      placeholder="Find your address"
      autocomplete="off"
    />
  </What3wordsAutosuggest>
</template>

<script lang="ts">
import {
  What3wordsAutosuggest,
} from "@what3words/vue-components";

export default {
  name: "Autosuggest",
  components: {
    What3wordsAutosuggest,
  },
  props: {
    callback: String,
    api_key: String,
    headers: String,
    base_url: String,
    name: String,
    initial_value: String,
    variant: String,
    typeahead_delay: Number,
    allow_invalid: Boolean,
    icon_visible: Boolean,
    icon_size: Number,
    icon_color: String,
    language: String,
    n_focus_results: Number,
    clip_to_country: String,
    clip_to_bounding_box: String,
    clip_to_circle: String,
    clip_to_polygon: String,
    return_coordinates: Boolean,
    onValue_changed: Function,
    onValue_valid: Function,
    onValue_invalid: Function,
    onDeselected_suggestion: Function,
    onSelected_suggestion: Function,
    onSuggestions_changed: Function,
    onCoordinates_changed: Function,
    on__hover: Function,
    on__focus: Function,
    on__blur: Function,
    on__error: Function,
  },
};
</script>

<style>
</style>
```



## Properties

| Property                        | Attribute                       | Description | Type                                                                | Default                                                            |
| ------------------------------- | ------------------------------- | ----------- | ------------------------------------------------------------------- | ------------------------------------------------------------------ |
| `api_key`                       | `api_key`                       |             | `string`                                                            | `DEFAULTS.emptyString`                                             |
| `api_version`                   | `api_version`                   |             | `ApiVersion.Version1 \| ApiVersion.Version2 \| ApiVersion.Version3` | `ApiVersion.Version3`                                              |
| `autosuggest_focus`             | `autosuggest_focus`             |             | `string`                                                            | `DEFAULTS.emptyString`                                             |
| `base_url`                      | `base_url`                      |             | `string`                                                            | `sdk.api.clients.autosuggest["_config"].host ?? DEFAULTS.base_url` |
| `callback`                      | `callback`                      |             | `string`                                                            | `DEFAULTS.emptyString`                                             |
| `clip_to_bounding_box`          | `clip_to_bounding_box`          |             | `string`                                                            | `DEFAULTS.emptyString`                                             |
| `clip_to_circle`                | `clip_to_circle`                |             | `string`                                                            | `DEFAULTS.emptyString`                                             |
| `clip_to_country`               | `clip_to_country`               |             | `string`                                                            | `DEFAULTS.emptyString`                                             |
| `clip_to_polygon`               | `clip_to_polygon`               |             | `string`                                                            | `DEFAULTS.emptyString`                                             |
| `headers`                       | `headers`                       |             | `string`                                                            | `DEFAULTS.headers`                                                 |
| `initial_value`                 | `initial_value`                 |             | `string`                                                            | `DEFAULTS.emptyString`                                             |
| `invalid_address_error_message` | `invalid_address_error_message` |             | `null \| string`                                                    | `DEFAULTS.null`                                                    |
| `language`                      | `language`                      |             | `string \| undefined`                                               | `undefined`                                                        |
| `n_focus_results`               | `n_focus_results`               |             | `number \| undefined`                                               | `undefined`                                                        |
| `name`                          | `name`                          |             | `string`                                                            | `DEFAULTS.name`                                                    |
| `options`                       | `options`                       |             | `CustomOption[]`                                                    | `[]`                                                               |
| `return_coordinates`            | `return_coordinates`            |             | `boolean`                                                           | `DEFAULTS.returnCoordinates`                                       |
| `strict`                        | `strict`                        |             | `boolean`                                                           | `DEFAULTS.true`                                                    |
| `typeahead_delay`               | `typeahead_delay`               |             | `number`                                                            | `DEFAULTS.typeaheadDelay`                                          |
| `variant`                       | `variant`                       |             | `Variant.DEFAULT \| Variant.INHERIT`                                | `DEFAULTS.variant`                                                 |


## Events

| Event                    | Description | Type                              |
| ------------------------ | ----------- | --------------------------------- |
| `__blur`                 |             | `CustomEvent<void>`               |
| `__error`                |             | `CustomEvent<{ error: Error; }>`  |
| `__focus`                |             | `CustomEvent<void>`               |
| `__hover`                |             | `CustomEvent<SuggestionPayload>`  |
| `coordinates_changed`    |             | `CustomEvent<CoordinatesPayload>` |
| `deselected_suggestion`  |             | `CustomEvent<SuggestionPayload>`  |
| `selected_custom_option` |             | `CustomEvent<CustomOption>`       |
| `selected_suggestion`    |             | `CustomEvent<SuggestionPayload>`  |
| `suggestions_changed`    |             | `CustomEvent<SuggestionsPayload>` |
| `suggestions_not_found`  |             | `CustomEvent<void>`               |
| `value_changed`          |             | `CustomEvent<InputValuePayload>`  |
| `value_invalid`          |             | `CustomEvent<InputValuePayload>`  |
| `value_valid`            |             | `CustomEvent<InputValuePayload>`  |


## Slots

| Slot                     | Description                                         |
| ------------------------ | --------------------------------------------------- |
| `"{ HTMLInputElement }"` | Optional input to override the default slot content |


## Dependencies

### Depends on

- [what3words-symbol](../what3words-symbol)

### Graph
```mermaid
graph TD;
  what3words-autosuggest --> what3words-symbol
  style what3words-autosuggest fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
