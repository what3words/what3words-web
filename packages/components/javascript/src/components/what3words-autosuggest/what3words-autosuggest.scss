@import url("https://fonts.googleapis.com/css?family=Source+Sans+Pro&display=swap");
@import "./flags.scss";

$prefix: "what3words-autosuggest";

// Set spinner/success state position based on LTR/RTL direction
:not([dir="rtl"]):not([dir="ltr"])
  what3words-autosuggest
  > .#{$prefix}
  > .#{$prefix}-input-wrapper
  > .#{$prefix}-state,
[dir="ltr"]
  what3words-autosuggest
  > .#{$prefix}
  > .#{$prefix}-input-wrapper
  > .#{$prefix}-state {
  right: 4px;
}
[dir="rtl"]
  what3words-autosuggest
  > .#{$prefix}
  > .#{$prefix}-input-wrapper
  > .#{$prefix}-state {
  left: 4px;
}

what3words-autosuggest {
  font-family: Source Sans Pro;
  font-style: normal;
  font-weight: normal;

  // Base settings
  & > .#{$prefix} {
    position: relative;
    display: flex;
    flex-direction: column;

    // INPUT
    & > .#{$prefix}-input-wrapper {
      display: flex;
      flex-direction: row;
      align-items: center;

      & > input {
        min-width: 1px;
      }

      & > .#{$prefix}-state {
        position: absolute;
        z-index: 100;
        width: 20px;
        height: 20px;
        margin-right: 0.25rem;
        content: " ";

        &.valid {
          background-image: url("data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0idXRmLTgiPz48c3ZnIHZlcnNpb249IjEuMSIgaWQ9IkxheWVyXzEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHg9IjBweCIgeT0iMHB4IiB2aWV3Qm94PSIwIDAgMzU2IDM1NiIgc3R5bGU9ImVuYWJsZS1iYWNrZ3JvdW5kOm5ldyAwIDAgMzU2IDM1NjsiIHhtbDpzcGFjZT0icHJlc2VydmUiPjxzdHlsZSB0eXBlPSJ0ZXh0L2NzcyI+LnN0MHtmaWxsOiMwMDk0NDQ7c3Ryb2tlOiMzOUI1NEE7c3Ryb2tlLW1pdGVybGltaXQ6MTA7fTwvc3R5bGU+PGcgaWQ9IndOUU5Tei50aWYiPjxnIGlkPSJYTUxJRF8xXyI+PGcgaWQ9IlhNTElEXzIyXyI+PHBhdGggaWQ9IlhNTElEXzI0XyIgY2xhc3M9InN0MCIgZD0iTTAsMTg2LjVjMC01LjcsMC0xMS4zLDAtMTdjMC42LTcuMiwxLjItMTQuNCwyLjUtMjEuNWM3LjctNDIuMywyNy42LTc3LjcsNjAuMy0xMDUuNmMyNS4xLTIxLjQsNTQuMS0zNC43LDg2LjctNDAuMWM2LjctMS4xLDEzLjQtMS42LDIwLjEtMi4zYzUuNywwLDExLjMsMCwxNywwYzYuNiwwLjYsMTMuMiwxLjEsMTkuOCwyLjJjMzkuMyw2LjYsNzMsMjQuMSwxMDAuNSw1M2MyNS40LDI2LjYsNDAuOSw1OC4yLDQ2LjksOTQuNWMxLjEsNi42LDEuNiwxMy4yLDIuMiwxOS44YzAsNS43LDAsMTEuMywwLDE3Yy0wLjYsNy4yLTEuMiwxNC40LTIuNSwyMS41Yy03LjcsNDIuMy0yNy42LDc3LjctNjAuMywxMDUuNmMtMjUuMSwyMS40LTU0LjEsMzQuNy04Ni43LDQwLjFjLTYuNywxLjEtMTMuNCwxLjYtMjAuMSwyLjNjLTUuNywwLTExLjMsMC0xNywwYy02LjYtMC42LTEzLjItMS4xLTE5LjgtMi4yYy0zOS4zLTYuNi03My0yNC4xLTEwMC41LTUzYy0yNS40LTI2LjYtNDAuOS01OC4yLTQ2LjktOTQuNUMxLjEsMTk5LjcsMC42LDE5My4xLDAsMTg2LjV6IE0zMzQuMywxNzguMUMzMzMuOCw5MS40LDI2NC45LDIyLDE3OC40LDIxLjhDOTEuNiwyMS42LDIyLjEsOTEsMjEuOCwxNzcuNUMyMS42LDI2NC41LDkxLDMzNCwxNzcuNywzMzQuMkMyNjQuNiwzMzQuMywzMzMuNywyNjQuOCwzMzQuMywxNzguMXoiLz48cGF0aCBpZD0iWE1MSURfMjNfIiBjbGFzcz0ic3QwIiBkPSJNMTQ3LjcsMjUzLjdjLTMuNywwLjItNi43LTEuMi05LjMtMy43Yy0xNy41LTE3LjYtMzUuMS0zNS4xLTUyLjYtNTIuN2MtNy4zLTcuNC00LjUtMTkuNCw1LjItMjJjNS4xLTEuNCw5LjYtMC4xLDEzLjUsMy43YzEyLjksMTIuOSwyNS44LDI1LjgsMzguNywzOC43YzQuNiw0LjYsNC42LDQuNiw5LjIsMGMzMi42LTMyLjYsNjUuMS02NS4xLDk3LjctOTcuN2M1LjUtNS41LDEzLjItNS45LDE4LjYtMWM1LjYsNS4xLDUuOCwxMy41LDAuMiwxOS4xYy03LDcuMS0xNC4xLDE0LjEtMjEuMiwyMS4yYy0zMC4xLDMwLjEtNjAuMiw2MC4yLTkwLjIsOTAuM0MxNTQuNywyNTIuNCwxNTEuNSwyNTQsMTQ3LjcsMjUzLjd6Ii8+PC9nPjwvZz48L2c+PC9zdmc+");
          background-position: center;
          background-repeat: no-repeat;
          background-size: 16px;
          align-self: center;
        }
        &.loading {
          display: block;
          border-radius: 50%;
          border: 1px solid #000;
          border-color: #000 transparent #000 transparent;
          animation: lds-dual-ring 1.2s linear infinite;
        }
      }
    }

    // SUGGESTIONS
    & > .#{$prefix}-suggestions {
      // Although these are re-invoked in the absolutes it is nice to have these as a parent setting which all child
      // elements should inherit or adhere to
      display: flex;
      flex-direction: column;
      cursor: pointer;

      &.hidden {
        display: none;
      }

      & > .#{$prefix}-items {
        position: absolute;
        display: flex;
        flex-direction: column;
        filter: drop-shadow(0px 0px 5px rgba(0, 0, 0, 0.2));
        border-bottom: 1px solid #7a7a7a;
        z-index: 500;

        & > .#{$prefix}-item {
          display: flex;
          flex-direction: column;
          align-items: flex-start;
          padding: 12px;
          background: #fff;
          box-shadow: inset 0px -1px 0px #f2f4f5;
          flex-grow: 1;
          cursor: pointer;

          // Hover and match state
          &.match {
            background: #dbeffa;
          }
          // Hover only
          &:hover {
            background: #fff;
            box-shadow: inset 0px 5000px 0px rgba(0, 83, 121, 0.08);
            transition: filter 100ms cubic-bezier(0.455, 0.03, 0.515, 0.955);
            &:active {
              filter: brightness(0.95);
            }
          }
          &:active,
          &.active {
            background: #fff;
            box-shadow: inset 0px 5000px 0px rgba(0, 83, 121, 0.15);
          }

          & > .#{$prefix}-address {
            pointer-events: none;
            display: flex;
            flex-direction: row;
            align-items: center;
            padding: 0px;
            position: static;
            height: 22px;
            left: 0px;
            top: 0px;
            flex: none;
            order: 0;
            align-self: stretch;
            flex-grow: 0;
            font-weight: 600;
            font-size: 17px;
            line-height: 22px;
            color: #0a3049;

            & > .#{$prefix}-words {
              width: 100%;
              white-space: nowrap;
              overflow: hidden;
              text-overflow: ellipsis;
              text-align: left;
            }
          }

          & > .#{$prefix}-nearest-place {
            display: flex;
            flex-direction: row;
            align-items: flex-start;
            padding: 0px 0px 0px 18px;
            align-self: stretch;

            & > .#{$prefix}-flag {
              margin-right: 3px;
              margin-left: 1px;
            }

            & > .#{$prefix}-nearest-place-text {
              display: flex;
              flex-grow: 1;
              justify-content: space-between;

              & > div:first-child {
                font-size: 13px;
                line-height: 18px;
                color: #525252;
                text-overflow: ellipsis;
              }

              & > .#{$prefix}-nearest-place-distance {
                display: flex;
                flex-wrap: nowrap;
                font-weight: 600;
                font-size: 13px;
                line-height: 18px;
                text-align: right;
              }
            }
          }
        }
      }
    }

    & > .#{$prefix}-error-wrapper {
      position: relative;
      display: flex;
      flex-direction: column;
      z-index: 300;

      & > .#{$prefix}-error {
        position: relative;
        display: flex;
        flex-direction: column;
        align-items: flex-start;
        width: 100%;
        top: -1px;
        border: 1px solid #c2c2c2;
        border-top: 1px solid #ed694e;
        background-color: #fff;

        & > .#{$prefix}-message {
          padding: 4px 16px 4px 16px;
          color: #ed694e;
          font-size: 14px;
          line-height: 20px;
        }
      }
    }

    // DEFAULT VARIANT STYLES
    &.default {
      & > .#{$prefix}-input-wrapper > input {
        display: flex;
        flex-grow: 1;
        padding: 12px 32px 12px 16px;
        background: #fff;
        color: #0a3049;
        border: 1px solid #c2c2c2;
        appearance: none;
        font-size: 18px;
        line-height: 23px;

        &:focus {
          outline: -webkit-focus-ring-color auto 5px;
        }
      }
      & > .#{$prefix}-error-wrapper > .#{$prefix}-error,
      & > .#{$prefix}-suggestions {
        margin-top: 0px;
      }
    }
  }
}

@keyframes lds-dual-ring {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
