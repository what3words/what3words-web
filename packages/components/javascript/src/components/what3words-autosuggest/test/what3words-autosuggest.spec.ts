import { newSpecPage } from "@stencil/core/testing";

import { What3wordsAutosuggest } from "../what3words-autosuggest";

describe("<what3words-autosuggest />", () => {
  it("renders", async () => {
    const { root } = await newSpecPage({
      components: [What3wordsAutosuggest],
      html: "<what3words-autosuggest><input></input></what3words-autosuggest>",
    });
    expect(root).toMatchSnapshot();
  });

  describe("when variant is provided", () => {
    it("renders", async () => {
      const value = "inherit";
      const { root } = await newSpecPage({
        components: [What3wordsAutosuggest],
        html: `<what3words-autosuggest variant=${value}><input></what3words-autosuggest>`,
      });
      expect(root).toMatchSnapshot();
    });
  });

  it("renders when wrapped an existing input element", async () => {
    const { root } = await newSpecPage({
      components: [What3wordsAutosuggest],
      html: '<what3words-autosuggest><input type="myinput" /></what3words-autosuggest>',
    });
    expect(root).toMatchSnapshot();
  });

  describe("when localised", () => {
    it("displays placeholder text as italian", async () => {
      const { root } = await newSpecPage({
        components: [What3wordsAutosuggest],
        html: "<what3words-autosuggest><input></what3words-autosuggest>",
        language: "it",
      });
      expect(root).toMatchSnapshot();
    });

    it("displays placeholder text as english if language is invalid", async () => {
      const { root } = await newSpecPage({
        components: [What3wordsAutosuggest],
        html: "<what3words-autosuggest><input></what3words-autosuggest>",
        language: "ww",
      });
      expect(root).toMatchSnapshot();
    });

    it("accepts language in uppercase", async () => {
      const { root } = await newSpecPage({
        components: [What3wordsAutosuggest],
        html: "<what3words-autosuggest><input></what3words-autosuggest>",
        language: "IT",
      });
      expect(root).toMatchSnapshot();
    });
  });
});
