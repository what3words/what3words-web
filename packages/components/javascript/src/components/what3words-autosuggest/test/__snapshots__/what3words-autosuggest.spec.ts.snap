// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<what3words-autosuggest /> renders 1`] = `
<what3words-autosuggest class="default">
  <!---->
  <div class="default what3words-autosuggest">
    <div class="what3words-autosuggest-input-wrapper" data-testid="input-wrapper">
      <input name="what3words_3wa" placeholder="e.g. ///lock.spout.radar" value="undefined">
      <span class="what3words-autosuggest-state" data-testid="state"></span>
    </div>
    <div class="hidden what3words-autosuggest-suggestions" data-testid="suggestions-wrapper">
      <div class="what3words-autosuggest-items"></div>
    </div>
  </div>
</what3words-autosuggest>
`;

exports[`<what3words-autosuggest /> renders when wrapped an existing input element 1`] = `
<what3words-autosuggest class="default">
  <!---->
  <div class="default what3words-autosuggest">
    <div class="what3words-autosuggest-input-wrapper" data-testid="input-wrapper">
      <input name="what3words_3wa" placeholder="e.g. ///lock.spout.radar" type="myinput" value="undefined">
      <span class="what3words-autosuggest-state" data-testid="state"></span>
    </div>
    <div class="hidden what3words-autosuggest-suggestions" data-testid="suggestions-wrapper">
      <div class="what3words-autosuggest-items"></div>
    </div>
  </div>
</what3words-autosuggest>
`;

exports[`<what3words-autosuggest /> when localised accepts language in uppercase 1`] = `
<what3words-autosuggest class="default">
  <!---->
  <div class="default what3words-autosuggest">
    <div class="what3words-autosuggest-input-wrapper" data-testid="input-wrapper">
      <input name="what3words_3wa" placeholder="p.es. ///mandali.avvera.sarò" value="undefined">
      <span class="what3words-autosuggest-state" data-testid="state"></span>
    </div>
    <div class="hidden what3words-autosuggest-suggestions" data-testid="suggestions-wrapper">
      <div class="what3words-autosuggest-items"></div>
    </div>
  </div>
</what3words-autosuggest>
`;

exports[`<what3words-autosuggest /> when localised displays placeholder text as english if language is invalid 1`] = `
<what3words-autosuggest class="default">
  <!---->
  <div class="default what3words-autosuggest">
    <div class="what3words-autosuggest-input-wrapper" data-testid="input-wrapper">
      <input name="what3words_3wa" placeholder="e.g. ///lock.spout.radar" value="undefined">
      <span class="what3words-autosuggest-state" data-testid="state"></span>
    </div>
    <div class="hidden what3words-autosuggest-suggestions" data-testid="suggestions-wrapper">
      <div class="what3words-autosuggest-items"></div>
    </div>
  </div>
</what3words-autosuggest>
`;

exports[`<what3words-autosuggest /> when localised displays placeholder text as italian 1`] = `
<what3words-autosuggest class="default">
  <!---->
  <div class="default what3words-autosuggest">
    <div class="what3words-autosuggest-input-wrapper" data-testid="input-wrapper">
      <input name="what3words_3wa" placeholder="p.es. ///mandali.avvera.sarò" value="undefined">
      <span class="what3words-autosuggest-state" data-testid="state"></span>
    </div>
    <div class="hidden what3words-autosuggest-suggestions" data-testid="suggestions-wrapper">
      <div class="what3words-autosuggest-items"></div>
    </div>
  </div>
</what3words-autosuggest>
`;

exports[`<what3words-autosuggest /> when variant is provided renders 1`] = `
<what3words-autosuggest class="inherit" variant="inherit">
  <!---->
  <div class="inherit what3words-autosuggest">
    <div class="what3words-autosuggest-input-wrapper" data-testid="input-wrapper">
      <input name="what3words_3wa" placeholder="e.g. ///lock.spout.radar" value="undefined">
      <span class="what3words-autosuggest-state" data-testid="state"></span>
    </div>
    <div class="hidden what3words-autosuggest-suggestions" data-testid="suggestions-wrapper">
      <div class="what3words-autosuggest-items"></div>
    </div>
  </div>
</what3words-autosuggest>
`;
